pkg/webhook/mutating/node/allocatable_mutator.go

```diff
@@ -41,154 +41,109 @@ func NewWebhookNodeAllocatableMutator() *WebhookNodeAllocatableMutator {
+	ratioValue, ok := node.Annotations[ratioKey]
 	if ok {
 		var newAllocatable, newCapacity *resource.Quantity
 		// get overcommit allocatable and capacity from annotation first
-		overcommitCapacity, ok := node.Annotations[consts.NodeAnnotationOvercommitCapacityMemoryKey]
+		capacity, ok := node.Annotations[capacityKey]
 		if ok {
-			quantity, err := resource.ParseQuantity(overcommitCapacity)
+			quantity, err := resource.ParseQuantity(capacity)
 			if err != nil {
-				klog.Error(err)
-			} else {
-				newCapacity = &quantity
-				klog.V(6).Infof("node %s mem capacity by annotation: %v", node.Name, newCapacity.String())
+				return err
 			}
+			newCapacity = &quantity
+			klog.V(6).Infof("node %s %s capacity by annotation: %v", node.Name, newCapacity.String(), newCapacity.String())
 		}
 
-		overcommitAllocatable, ok := node.Annotations[consts.NodeAnnotationOvercommitAllocatableMemoryKey]
+		allocatable, ok := node.Annotations[allocatableKey]
 		if ok {
-			quantity, err := resource.ParseQuantity(overcommitAllocatable)
+			quantity, err := resource.ParseQuantity(allocatable)
 			if err != nil {
-				klog.Error(err)
-			} else {
-				newAllocatable = &quantity
-				klog.V(6).Infof("node %s mem allocatable by annotation: %v", node.Name, newAllocatable.String())
+				return err
 			}
+			newAllocatable = &quantity
+			klog.V(6).Infof("node %s %s allocatable by annotation: %v", node.Name, newCapacity.String(), newAllocatable.String())
 		}
 
+		// calculate allocatable and capacity by overcommit ratio
 		if newAllocatable == nil || newCapacity == nil {
-			memoryOvercommitRatio, err := memOvercommitRatioValidate(node.Annotations)
+			ratio, err := validateOvercommitRatio(node.Annotations, ratioKey)
 			if err != nil {
-				klog.Errorf("node %s %s validate fail, value: %s, err: %v", node.Name, consts.NodeAnnotationMemoryOvercommitRatioKey, memoryOvercommitRatioValue, err)
+				return fmt.Errorf("node %s %s validate fail, value: %s, err: %v", node.Name, ratioKey, ratioValue, err)
 			} else {
-				if memoryOvercommitRatio > 1.0 {
-					allocatable := node.Status.Allocatable.Memory()
-					capacity := node.Status.Capacity.Memory()
-					allocatableByOvercommit := native.MultiplyResourceQuantity(core.ResourceMemory, *allocatable, memoryOvercommitRatio)
-					capacityByOvercommit := native.MultiplyResourceQuantity(core.ResourceMemory, *capacity, memoryOvercommitRatio)
+				if ratio > 1.0 {
+					allocatable := node.Status.Allocatable[rn]
+					capacity := node.Status.Capacity[rn]
+					allocatableByOvercommit := native.MultiplyResourceQuantity(rn, allocatable, ratio)
+					capacityByOvercommit := native.MultiplyResourceQuantity(rn, capacity, ratio)
 					newAllocatable = &allocatableByOvercommit
 					newCapacity = &capacityByOvercommit
 					klog.V(6).Infof("node %s %s capacity: %v, allocatable: %v, newCapacity: %v, newAllocatable: %v",
-						node.Name, core.ResourceMemory,
+						node.Name, newCapacity.String(),
```

Comment:
The second param should be rn.String() ?

Reply:
done